crates/sui-graphql-rpc/src/types/owner.rs

```diff
@@ -99,6 +99,18 @@ pub(crate) struct Owner {
     pub address: SuiAddress,
 }
 
+/// The basic unit of storage in Sui is object. In contrast to many other blockchains,
+/// where storage is centered around address and each address contains a key-value store,
+/// Sui's storage is centered around objects. Sui objects have one of the following primary states:
+///
+///    Immutable - the object cannot be modified.
+///    Mutable - the object can be changed.
+///
+/// Further, mutable objects are divided into these categories:
+///
+///    Owned - the object can be modified only by its owner.
+///    Shared - the object can be modified by anyone.
+/// Immutable objects do not need this distinction because they have no owner.
```

Comment:
Words are really hard so I'm not phrasing it in a way I like, but for Owner, I'd say it's an abstraction over Address and Object so that someone who doesn't know what the type of owner is can still look up shared fields like the owner's dynamic fields, balances, etc.

Reply:
What about this:

> An `Owner` is an entity that can own an object.  Each `Owner` is identified by a `SuiAddress` which represents either an `Address` (corresponding to a public key of an account) or an `Object`, but never both (it is not known up-front whether a given `Owner` is an `Address` or an `Object`).

Agreed that it's not appropriate to re-use the `Object` docs here -- they are different things.

Also we should document `as_address` and `as_object` (it also looks like we need to implement `as_object`...